### ISSUE TITLE
`taylorinteg` not working properly with BigFloats

### ISSUE THREAD

**Issue by PerezHz** [PerezHz/TaylorIntegration.jl#27]
On current master, `taylorinteg` breaks down when using BigFloats, if the `maxsteps` keyword argument is greater than the actual number of steps taken during a given integration:

```julia
julia> using TaylorIntegration

julia> function harmosc!(t, x, dx) #the harmonic oscillator ODE
           dx[1] = x[2]
           dx[2] = -x[1]
           nothing
       end
harmosc! (generic function with 1 method)

julia> @time tv, xv = taylorinteg(harmosc!, [0, -1], 0, 20BigFloat(pi), 90, 1.0e-80, maxsteps=15 );
WARNING: Maximum number of integration steps reached; exiting.
  0.663779 seconds (452.26 k allocations: 25.480 MiB, 9.63% gc time)

julia> tv[end] == 20BigFloat(pi)
true

julia> @time tv, xv = taylorinteg(harmosc!, [0, -1], 0, 20BigFloat(pi), 90, 1.0e-80, maxsteps=16 );
ERROR: UndefRefError: access to undefined reference
Stacktrace:
 [1] transpose_f!(::Base.#transpose, ::Array{BigFloat,2}, ::Array{BigFloat,2}) at ./linalg/transpose.jl:54
 [2] transpose(::Array{BigFloat,2}) at ./linalg/transpose.jl:121
 [3] #taylorinteg#4(::Int64, ::Function, ::Function, ::Array{BigFloat,1}, ::BigFloat, ::BigFloat, ::Int64, ::BigFloat) at /Users/Jorge/TaylorIntegration.jl/src/explicitode.jl:453
 [4] (::TaylorIntegration.#kw##taylorinteg)(::Array{Any,1}, ::TaylorIntegration.#taylorinteg, ::Function, ::Array{BigFloat,1}, ::BigFloat, ::BigFloat, ::Int64, ::BigFloat) at ./<missing>:0
 [5] #taylorinteg#3(::Int64, ::Function, ::Function, ::Array{Int64,1}, ::Int64, ::BigFloat, ::Int64, ::Float64) at /Users/Jorge/TaylorIntegration.jl/src/explicitode.jl:408
 [6] (::TaylorIntegration.#kw##taylorinteg)(::Array{Any,1}, ::TaylorIntegration.#taylorinteg, ::Function, ::Array{Int64,1}, ::Int64, ::BigFloat, ::Int64, ::Float64) at ./<missing>:0
```

In the example above, if `maxsteps=15` or less, then the answer is returned correctly, but if `maxsteps` is greater than 15, then it just breaks down. I think this is related with how `transpose` handles matrices with `#undef` components, when the elements of the matrix are BigFloats. When working with Float64, then the problem doesn't occur:

```julia
julia> @time tv, xv = taylorinteg(harmosc!, [0, -1], 0, 20pi, 25, 1.0e-20, maxsteps=200 );
  0.359674 seconds (200.31 k allocations: 10.576 MiB)
```

**Comment by lbenet** [comment]
Can you isolate the problem without TaylorIntegration? I mean, create such a vector, fill it partially, and transpose it. Perhaps this should be posted directly to Julia.

**Comment by lbenet** [comment]
I think the problem can be solved replacing the lines which look like:

```julia
    return view(tv,1:nsteps), view(transpose(xv),1:nsteps,:)
```

by

```julia
    return view(tv,1:nsteps), transpose(view(xv,:,1:nsteps))
```

Maybe it impacts in performance, but my tests make it difficult to decide...

**Comment by PerezHz** [comment]
> Can you isolate the problem without TaylorIntegration? I mean, create such a vector, fill it partially, and transpose it. Perhaps this should be posted directly to Julia.

Following your recommendation, I think an example of this would be:

```julia
julia> A = Array{Float64}(2,3)
2×3 Array{Float64,2}:
 2.34463e-314  2.34463e-314  2.34463e-314
 2.34463e-314  2.34463e-314  2.34463e-314

julia> A[1,:] .= rand(3)
3-element SubArray{Float64,1,Array{Float64,2},Tuple{Int64,Base.Slice{Base.OneTo{Int64}}},true}:
 0.171995
 0.35663 
 0.263829

julia> transpose(A)
3×2 Array{Float64,2}:
 0.171995  2.34463e-314
 0.35663   2.34463e-314
 0.263829  2.34463e-314

julia> B = Array{BigFloat}(2,3)
2×3 Array{BigFloat,2}:
 #undef  #undef  #undef
 #undef  #undef  #undef

julia> B[1,:] .= BigFloat.(rand(3))
3-element SubArray{BigFloat,1,Array{BigFloat,2},Tuple{Int64,Base.Slice{Base.OneTo{Int64}}},true}:
 5.221675004832033728519036230864003300666809082031250000000000000000000000000000e-01
 9.018393963488640086012537722126580774784088134765625000000000000000000000000000e-01
 3.842859182000140094714879523962736129760742187500000000000000000000000000000000e-01

julia> transpose(B)
ERROR: UndefRefError: access to undefined reference
Stacktrace:
 [1] transpose_f!(::Base.#transpose, ::Array{BigFloat,2}, ::Array{BigFloat,2}) at ./linalg/transpose.jl:54
 [2] transpose(::Array{BigFloat,2}) at ./linalg/transpose.jl:121
```

Do you think it would be worthwhile to report this issue to the corresponding Julia org?

**Comment by PerezHz** [comment]
I agree that #29 could be a workaround!

**Comment by PerezHz** [comment]
> Maybe it impacts in performance, but my tests make it difficult to decide...

Do you want me to check some performance benchmarks?

**Comment by lbenet** [comment]
Yes, I think it is a good idea to perform some benchmarks.

Regarding the issue with the `transpose`, I do think this is an issue for Julia directly. Yet, as you can see in your example, when you allocate the matrix of `Float64`, it is filled with whatever `Float64`s, which is not the case for `BigFloat`s. That is the reason that the error does not occur with the `Float64` matrix.

**Comment by PerezHz** [comment]
Closed by #33
